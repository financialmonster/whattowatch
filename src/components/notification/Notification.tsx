import React, { FC, memo } from 'react';
import styled, { keyframes } from 'styled-components';

type TNotificationProps = {
    children: string;
}

const fadeInOut = keyframes`
    0% {
        opacity: 0;
    }

    50% {
        opacity: 1;
    }

    100% {
        opacity: 0;
    }
`;

const NotificationWrapper = styled.div`
    position: fixed;
    right: 0;
    bottom: 0;
    min-height: 50px;
    width: 300px;
    outline: 1px solid red;
    padding: 10px 20px;
    background-color: black;
    opacity: 0;
    animation: ${fadeInOut} linear 5s;
`;

export const Notification: FC<TNotificationProps> = memo(({ children }) => {
    return (
        <NotificationWrapper>
            {children}
        </NotificationWrapper>
    );
});
