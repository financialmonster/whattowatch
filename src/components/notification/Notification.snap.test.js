import React from 'react';
import renderer from 'react-test-renderer';

import { Notification } from './Notification';

it(`Notification: correctly rendered`, () => {
    const tree = renderer.create(
        <Notification>
            Test
        </Notification>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
