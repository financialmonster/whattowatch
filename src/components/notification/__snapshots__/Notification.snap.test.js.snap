// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification: correctly rendered 1`] = `
<div
  className="sc-AxjAm PAzfP"
>
  Test
</div>
`;
